heka/_service.sls

```
@@ -145,9 +141,13 @@ heka_{{ service_name }}_service:
 {# Load the other services' support metadata from salt-mine #}
 
 {%- for node_name, node_grains in salt['mine.get']('*', 'grains.items').iteritems() %}
-{%- if node_grains.heka is defined %}
-{%- set service_grains = salt['grains.filter_by']({'default': service_grains}, merge=node_grains.heka) %}
-{%- endif %}
+{%- if node_grains.heka is defined and node_grains.heka is mapping %}
```

Minor, but I would not bother checking that `node_grains.heka` is a mapping.

---

why not ? I prefer check this is a dict() befor iterate over it.

---

I'd just rather fail if heka is not a mapping rather than silently skipping it.